RoslynPluginGenerator/NuGet/NuGetPackageHandler.cs

```diff
@@ -17,60 +16,17 @@ namespace SonarQube.Plugins.Roslyn
 {
     public class NuGetPackageHandler : INuGetPackageHandler
     {
-        private readonly string packageSource;
+        private readonly IPackageRepository remoteRepository;
         private readonly string localCacheRoot;
         private readonly Common.ILogger logger;
 
-        private class NuGetLoggerAdapter : NuGet.ILogger
+        public NuGetPackageHandler(IPackageRepository remoteRepository, string localCacheRoot, Common.ILogger logger)
```

Comment:
There were three constructors; I removed two of them and fixed up the tests and product code.